code/confounds/confounds.py

```diff
@@ -0,0 +1,262 @@
+import pandas as pd
+from pathlib import Path
+
+
+def get_confounds_scanstsv(dataset_path="/data/datasets/hcph-dataset"):
+    """
+    Extract and process confound information from a `scans.tsv` file.
+
+    This function reads a `scans.tsv` file from the specified dataset path, processes
+    its contents to extract relevant confound information, and returns a DataFrame
+    with additional metadata columns.
+
+    Parameters:
+    -----------
+    dataset_path : str, optional
+        The path to the dataset directory containing the `scans.tsv` file.
+        Default is "/data/datasets/hcph-dataset".
+
+    Returns:
+    --------
+    pd.DataFrame
+        A pandas DataFrame containing the processed confound information with the following columns:
+        - `filename`: The original filename of the scan.
+        - `acq_time`: The acquisition time of the scan.
+        - `subject`: The subject identifier extracted from the filename.
+        - `session`: The session identifier extracted from the filename.
+        - `modality`: The modality of the scan (e.g., anat, func) extracted from the filename.
+        - `task`: The task identifier extracted from the filename.
+        - `pe_dir`: The phase encoding direction extracted from the filename.
+        - `day_of_week`: The day of the week extracted from the acquisition time.
+        - `time_of_day`: The rounded time of day extracted from the acquisition time.
+    """
+
+    ## Read the scans.tsv file
+    dataset_path = Path(dataset_path)
+    confounds_df = pd.read_csv(dataset_path / "scans.tsv", sep="\t")
+    confounds_df.drop(columns=["randstr"], inplace=True)
+
+    # Extract entities
+    # Each echo and magnitude and phase part of each functional scans are recorded on separate line,
+    # but we have only one final image so keep only one of the identical acq_time
+    confounds_df = confounds_df.drop_duplicates(subset=["acq_time"])
+    confounds_df = confounds_df.assign(
+        subject=confounds_df["filename"].str.extract(r"sub-(\d+)/"),
+        session=confounds_df["filename"].str.extract(r"ses-(\w+)/"),
+        modality=confounds_df["filename"].str.split("/").str[2],
+        task=confounds_df["filename"].str.extract(r"task-(\w+)_"),
+        pe_dir=confounds_df["filename"].str.extract(r"dir-(\w+)_"),
+    )
+
+    # From the acq_time column, extract the day of week and time of day
+    confounds_df = confounds_df.assign(
+        datetime=pd.to_datetime(confounds_df["acq_time"])
+    )
+    confounds_df = confounds_df.assign(
+        day_of_week=confounds_df["datetime"].dt.day_name(),
+        time_of_day=confounds_df["datetime"].dt.round("H").dt.time,
+    )
+    confounds_df.drop(columns=["datetime"], inplace=True)
+
+    return confounds_df
+
+
+def get_iqms(
+    iqms_path="/data/derivatives/hcph-mriqc/group_dwi.tsv", iqm_of_interest=["fd_mean"]
```

Comment:
Not a good idea to make the critical parameter optional
```suggestion
    iqms_path, iqm_of_interest=["fd_mean"]
```